python/setup.py

```diff
@@ -120,7 +120,7 @@ def config_cython():
     install_requires=[
         'numpy>=1.14.0',
         'scipy>=1.1.0',
-        'networkx>=2.1',
+        'networkx>=2.1,<2.4',
```

Comment:
Let's use the newer networkx version. That means we need to drop the usage of stale APIs.

Reply:
It think drop the usage of stale APIs is better. 
But I think we should find a way to avoid annoying error messages in lots of our examples before we fix them. This is not friendly to new DGL users when they run the tutorial codes and have misleading errors like this.
At least we should add some Doc announce maybe.